src/app/games/yinsh/yinsh.component.ts

```diff
@@ -247,38 +275,21 @@ export class YinshComponent extends HexagonalGameComponent<YinshRules,
             this.lastMoved = [move.start];
         } else {
             this.lastMoved = this.coordsBetween(move.start, move.end.get());
-            const nothingSelectedThisTurn: boolean =
-                this.currentCapture.isAbsent() &&
-                this.initialCaptures.length === 0 &&
-                this.moveStart.isAbsent();
-            this.lastCaptured = [];
-            move.initialCaptures.forEach((c: YinshCapture) => this.showLastMoveCapture(c, nothingSelectedThisTurn));
-            move.finalCaptures.forEach((c: YinshCapture) => this.showLastMoveCapture(c, nothingSelectedThisTurn));
+            this.lastCapture = MGPOptional.empty();
+            move.initialCaptures.forEach((c: YinshCapture) => this.lastCapture = MGPOptional.of(c));
+            move.finalCaptures.forEach((c: YinshCapture) => this.lastCapture = MGPOptional.of(c));
```

Comment:
sang de courge, t'as tu vu c'que tu laisse passer lo !?

Reply:
bah eh, stok ça ! une lettre sur une ligne c'est acceptaap

Reply:
mais setted N fois de suite un optional à truc-de-N c'est débildoss ! Soit on prend le dernier, soit c'est TUNE LISSS !

Reply:
haha binwinw, j'ai pas vu même quand tu me le pointe du doigt ! Un assignement dans un forEach ça devrait être suspicieux pour sûr !
Quid de `this.lastCaptures.concat(move.initialCaptures)` ou du genre, idem pour le second ?